File: 9_9_19/binary_tree_level_order_traversal.py
# Runtime O(n)
# Memory O(n)

def levelOrder(root):
    if root is None: return []
    fill = []
    levelHelper(root, 0, fill)
    return fill

def levelHelper(node, level, fill):
    if len(fill) == level:
        fill.append([])
    fill[level].append(node.val)
    if node.left: levelHelper(node.left, level + 1, fill)
    if node.right: levelHelper(node.right, level + 1, fill)

File: 9_9_19/binary_tree_level_order_traversal.js
// Runtime O(n)
// Memory O(n)

var levelOrder = function(root) {
    if (!root) return []
    const fill = [];
    const helper = (node, level) => {
        if (fill.length === level) {
            fill.push([]);
        }
        fill[level].push(node.val);
        if (node.left) helper(node.left, level + 1);
        if (node.right) helper(node.right, level + 1);
    }
    helper(root, 0)
    return fill;
};